### DISCUSSION TITLE
Antd components with no types

### DISCUSSION THREAD

**Discussion by sardor01** [nuxt/nuxt#19016] [Questions] [answered]
### Environment

- Operating System: `Linux`
- Node Version:     `v18.13.0`
- Nuxt Version:     `3.2.0`
- Nitro Version:    `2.2.1`
- Package Manager:  `pnpm@7.27.0`
- Builder:          `vite`
- User Config:      `ssr`, `css`, `modules`, `typescript`, `postcss`
- Runtime Modules:  `@pinia/nuxt@0.4.6`, `@nuxtjs/i18n@8.0.0-beta.9`
- Build Modules:    `-`

### Reproduction

https://github.com/Sardor01/nux3-admin-antd

### Describe the bug

I installed ant-design-vue and want to use it inside any vue file. When hover over any ant-design-vue component, the type is `any`. I can't find any way of adding types to the project. I thought about extending `.nuxt/components.d.ts`, but it's impossible since it's auto generated. How should I make it **typesafe**?

### Additional context

![image](https://user-images.githubusercontent.com/58367523/218652291-3b4850d8-5923-472e-88c6-d6dfcacf079a.png)

**Comment by danielroe** [comment, marked as answer]
You can remove the plugin where you register every component and instead add this module to you nuxt config:

```js
import * as AntD from 'ant-design-vue'
import { addComponent } from '@nuxt/kit'

export default defineNuxtConfig({
  modules: [
    function (options, nuxt) {
      for (const key in AntD) {
        if (['version', 'install'].includes(key)) continue
        await addComponent({
          filePath: 'ant-design-vue',
          name: `A${key}`,
          export: key
        })
      }
    },
  },
})
```
